src/d3a/models/market/one_sided.py

```diff
@@ -173,12 +174,12 @@ def determine_offer_price(self, energy_portion, energy, trade_rate,
                 energy, trade_rate, energy_portion, orig_offer_price
             )
         else:
-            revenue, fees, trade_rate_incl_fees = \
+            revenue, grid_fee_rate, trade_rate_incl_fees = \
                 GridFees.calculate_trade_price_and_fees(
                     trade_bid_info, self.transfer_fee_ratio
                 )
-            self.market_fee += fees
-            return energy * (trade_rate_incl_fees - fees) \
+            self.market_fee += grid_fee_rate * energy_portion * energy
+            return grid_fee_rate, energy * (trade_rate_incl_fees - grid_fee_rate) \
```

Comment:
The return values in the if block are not consistent, if the if block is True 2 values are returned, if it is False only one value is returned.

Reply:
Ah you are right, the if block applies to the second argument, not the tuple. All good